satpy/writers/__init__.py

```diff
@@ -381,6 +381,46 @@ def to_image(dataset, copy=False, **kwargs):
         return XRImage(dataset)
 
 
+def compute_writer_results(results):
+    """Compute all the given dask graphs `results` so that the files are
+    saved.
+
+    Args:
+        results (iterable): Iterable of dask graphs resulting from call to
+                            `scn.save_datasets(..., compute=False)
+    """
+    sources = []
+    targets = []
+    delayeds = []
+    for res in results:
+        if isinstance(res, tuple):
+            # source, target to be passed to da.store
+            # Check for empty results
+            if len(res) == 0:
+                continue
+            num = len(res[0])
+            if num == 0:
+                continue
+            for i in range(num):
+                sources.append(res[0][i])
+                targets.append(res[1][i])
```

Comment:
I'm not sure it matters performance-wise but couldn't this be replaced with `extend` and `zip`? Then you could probably also remove the check for zero length.